### ISSUE TITLE
Noob question? Namespace not found from code after successful install otherwise (installed via the package manager)

### ISSUE THREAD

**Issue by prenex** [atteneder/glTFast#178]
Hi!

I have no idea why this is, but after (hopefully properly) following the steps for installation, unity and compiler does not seem to see the gltFast package at all.

Quite interestingly I can see the code they do not find in the "Project" view (see attached image about this and error message).

See:
https://pasteboard.co/K1AEBuC.png

Also quite interestingly I can even use "Add Component" to add the "GltfAsset script!!! (see other attached image)
See:
https://pasteboard.co/K1AF4xFO.png

What I am doing wrong??? How make unity / VS / whatever see the files in question?

It can be that this is a noob question because I did not really used scripts from packages like this before.

I am using Unity 2019.4.26f and cannot update further for now because of other reasons unrelated for this but affecting the project. I do not know if it counts though.

If this is just some noob question regarding unity I hope it helps others too :-)
prenex

**Comment by atteneder** [comment]
Hi @prenex ,

If your script is part of (in the same folder as) an assembly definition, you should check if your assembly definition references the glTFast runtime assembly. Read more about [assembly definitions](https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html).

Otherwise it would help if I can see some demo code that's not working.

**Comment by prenex** [comment]
In the meantime I debugged the stuff like crazy and went a big circle:

1.) Tried all the usual "unity-ism": reimport, retry, restart, whatever
2.) Grabbed all your source codes from the package to some dir(*).
3.) Removed the package via package manager.
4.) Manually copied your sources over into some scripts subfolder somewhere from (TMP)
5.) It then complained about Unity.Mathematics in a similar way - error now coming from your sources.
6.) Added that one via package manager. No error anymore from your sources as they find it (hurray)
7.) Tried to reference glTFast namespace now from code - it still fails ?!?!
8.) I see in VS there shown to have various projects for the various asmdef files you had and I copied over from (TMP) temporary location.
9.) Remove all of these again
10.) Delete asmdef files and only keep .meta and .cs files from among yours at (TMP)
11.) Add everything below scripts from (TMP)
12.) Again compile error because missing Unity.Mathematics ???
13.) Now I realize it is already installed via package manager - hmmm interesting. Also remove/reinstall etc. do not help
14.) I REALIZE THAT: my more main project asmdef file now has:     "references": [],
15.) I change it to      "references": ["Unity.Mathematics"],
16.) Now your code added from (TMP) does not complain anymore for this. When I try to refer to namespace it works, but there are random compile error when some of your classes not seeing some of your other ones.
17.) I realize that this is likely because asmdef files have been deleted by me
18.) I again remove all my changes
19.) Add your stuff again "normall" through package manager
20.) I change my own "main" asmdef file to have these:      "references": ["Unity.Mathematics", "glTFast"],

and now? Now it seems it just works! At least builds, but I did not really try it much yet, but I guess the issue I originally had is gone.

**Comment by prenex** [comment]
TL;DR: For some reason unity might not add a reference automatically, so all I had to do is to look at my own main asmdef file

and change this:
`"references": [],`
into this:
`"references": ["Unity.Mathematics", "glTFast"],`

manually for some reason. I am not sure if this is expected or is some unity glitch, but you can avoid the above lengthy debugging with this I hope in case any reader has error similar to these.

**Comment by prenex** [comment]
Will now test how loading works and if it also works on some device :-)

**Comment by prenex** [comment]
Works fully - both on the PC and my "device" except that I have now unrelated issues with material not showing up on the device (but it shows on a PC - likely some of the shaders, but this is separate issue, closing this one)

**Comment by atteneder** [comment]
Great that you found the solution!

Regarding materials "not working" in builds: please make sure to read [THIS SECTION](https://github.com/atteneder/glTFast#materials-and-shader-variants) very carefully.

hth

**Comment by prenex** [comment]
Thank you for that tip!

What I did was to add the gltf/PbrMetallicRoughness and gltf/PbrMetallicGlossiness and also the "unlit" shader to the "project settings / graphics / always included shaders" already, but build is still going on and going on...

As you describe it now, there are many-many variants. It was not sure for me that this glTFast works with shaders and thought maybe one could get away with directly reusing standard ones in unity. I see this is likely much more standars compliant of course (I actually have developed a C++ gtlf I/O myself not long ago as a 3dsmax plugin - even named some of the classes just like you did haha - fun fact).

I am not sure if I can (or should) really constrain what combinations of models / and their shader needs will be and not fully clear are the steps you outline there for making this process faster, but finally as the build finished, I might keep just having them all if this is how I have them all.

If I would just use gltFast for streaming models that a controlled team makes or always save them in some architecture in some materials what you write on your link would likely hugely help though so thanks for mentioning these (and also giving hope this is why it was not working - here it is late night and have been debugging move semantics issues in C++, then hunting down extra info for tender grant process administration and stuff, so not in best brain capacity at right now.

Your imput and fast response is really helpful and encouraging though! Nice project overall!
